src/librustc/middle/resolve_lifetime.rs

```diff
@@ -1441,6 +1446,18 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             .collect();
 
         let next_early_index = index + generics.ty_params().count() as u32;
+        let mut arg_lifetimes = vec![];
+        for input in &decl.inputs {
```

Comment:
This vector is only used in the case of lint/error, right? I'm not sure how I feel about building it eagerly -- maybe we can convert this into a closure or a trait or something?

Reply:
Makes sense, I'll see what I can do about it.